ecs-cli/modules/cli/logs/logs_app.go

```diff
@@ -288,3 +289,31 @@ func getContainerLogConfig(containerDef *ecs.ContainerDefinition) (*logConfigura
 func logConfigMisMatchError(taskDef *ecs.TaskDefinition, fieldName string) error {
 	return fmt.Errorf("Log Configuration Field %s mismatches in at least one container definition in %s. Use the --%s option to query logs for an individual container.", fieldName, aws.StringValue(taskDef.TaskDefinitionArn), flags.ContainerNameFlag)
 }
+
+/* Create Logs */
+
+// CreateLogGroups creates any needed log groups for the task definition to use CloudWatch Logs
+func CreateLogGroups(taskDef *ecs.TaskDefinition, logClientFactory cwlogsclient.LogClientFactory) error {
+	for _, container := range taskDef.ContainerDefinitions {
+		logConfig, err := getContainerLogConfig(container)
+		if err != nil {
+			return err
+		}
+		region := aws.StringValue(logConfig.logRegion)
+		client := logClientFactory.Get(region)
+		err = client.CreateLogGroup(logConfig.logGroup)
+		if err != nil {
+			if aerr, ok := err.(awserr.Error); ok {
+				if aerr.Code() == cloudwatchlogs.ErrCodeResourceAlreadyExistsException {
+					// If the log group already exists warn the user but don't fail the command
+					logrus.Warnf("Failed to create log group %s in %s: %s", aws.StringValue(logConfig.logGroup), region, aerr.Message())
+					return nil
+				}
+			}
```

Comment:
Should we not still return other errors that aren't specifically ErrCodeResourceAlreadyExistsExceptions here?

Reply:
Good catch! Yes!